js/plugin-loader.js

```diff
@@ -644,7 +655,12 @@ PluginLoader.prototype = {
     }
     this.plugins = this._toposortPlugins(pluginContext.plugins, this.pluginMap);
 //    bootstrapLogger.warn('pluginMap empty (plugin-loader.js line530)='
-//        + JSON.stringify(this.pluginMap));    
+//        + JSON.stringify(this.pluginMap));  
+    
+    // Share with index.js the amount of plugins that should load
+    this.emit('givePluginAmount', {
```

Comment:
This doesn't look like an event either. In fact, the explicit goal of this design was to make both filesystem and dynamic plugins "first-class citizens" and thus we *can't* know the number of plugins in advance.

If an external caller needs to know whether we're done with loading the filesystem plugins, we can dispatch an event once that's done (and we're now waiting for external plugins)

Also, I know it's just a comment, but even mentioning another module's name here might indicate anti-modular inclinations: we must not care who the listener is, we just implement the contact for the plugin loader. Whether 'index.js' or 'foo.js' or whatever uses us, that shouldn't affect us, we should just do our thing.

Reply:
You must know the number of initial plugins for certain things (import resolution, eureka handshake) so an event was created. Dynamic adding should be delayed until after this point to avoid a problem in counting but it happens very quickly currently so it hasnt yet posed a problem.